modules/platform/rest-api/nuxeo-rest-api-server/src/main/java/org/nuxeo/ecm/restapi/server/jaxrs/adapters/PaginableAdapter.java

```diff
@@ -114,22 +118,27 @@ protected DocumentModel getSearchDocument() {
         PageProviderService pps = Framework.getService(PageProviderService.class);
         Map<String, Serializable> props = new HashMap<>();
         props.put(CoreQueryDocumentPageProvider.CORE_SESSION_PROPERTY, (Serializable) ctx.getCoreSession());
-        PageProvider<T> pp = (PageProvider<T>) pps.getPageProvider("", ppDefinition, getSearchDocument(), null,
-                pageSize, currentPageIndex, props, getParams());
+        List<SortInfo> sortersInfos = null;
         if (!StringUtils.isBlank(sortBy)) {
             String[] sorts = sortBy.split(",");
             String[] orders = null;
             if (!StringUtils.isBlank(sortOrder)) {
                 orders = sortOrder.split(",");
             }
-            // clear potential default sort infos first
-            pp.setSortInfos(null);
-            for (int i = 0; i < sorts.length; i++) {
-                String sort = sorts[i];
-                boolean sortAscending = orders != null && orders.length > i && "asc".equals(orders[i].toLowerCase());
-                pp.addSortInfo(sort, sortAscending);
+
+            if (ArrayUtils.isNotEmpty(sorts)) {
```

Comment:
Is this really useful? If `sortBy` isn't blank, then `sortBy.split(",")` should always return a non empty array no?

Reply:
Yep, I added the check to avoid passing an empty sorters to underline provide have a look at [here](https://github.com/nuxeo/nuxeo/blob/master/modules/platform/nuxeo-platform-query-api/src/main/java/org/nuxeo/ecm/platform/query/core/PageProviderServiceImpl.java#L115) From what I saw is if we pass empty array it will consider it as an empty sorter and it will end by not sorting (will not use the default ones).

Reply:
But I re-check  a second time we can remove it, thx